src/components/utils/useIsFirstRender.ts

```diff
@@ -1,14 +1,17 @@
-import {useRef} from 'react';
+import React from 'react';
 
-const useIsFirstRender = () => {
-  const isFirstRender = useRef(true);
+export const useIsFirstRender = () => {
+  const [isFirstRender, setIsFirstRender] = React.useState(true);
 
-  if (isFirstRender.current === true) {
-    isFirstRender.current = false;
-    return true;
-  }
+  React.useLayoutEffect(() => {
```

Comment:
hey @clxandstuff , we need isomorphic version here

Reply:
Done